packages/dht/src/connection/connectivityChecker.ts

```diff
@@ -78,9 +80,15 @@ export const sendConnectivityRequest = async (
                     if (message.body.oneofKind === 'connectivityResponse') {
                         logger.debug('ConnectivityResponse received: ' + JSON.stringify(Message.toJson(message)))
                         const connectivityResponseMessage = message.body.connectivityResponse
+                        const remoteVersion = connectivityResponseMessage.version
                         outgoingConnection!.off('data', listener)
                         clearTimeout(timeoutId)
-                        resolve(connectivityResponseMessage)
+                        // remoteVersion can be undefined if the remote peer is running an older version
+                        if (remoteVersion !== undefined && isCompatibleVersion(localVersion, remoteVersion)) {
```

Comment:
Are you sure it is `undefined` and not empty string `''` when remote peer is running an older version? Asking because the `ConnectivityResponse#ipAddress` field apparently gets default value 0 when not present? So is there a same kind of "assign default value if not present" logic here w.r.t. protobuf behavior?

Reply:
Good point, here it is indeed an empty string instead of undefined as well. So the undefined check is likely completely unnecessary since the isCompatibleVersion correctly handles empty strings in this case